go-controller/pkg/metrics/ovnkube_controller.go

```diff
@@ -619,6 +653,56 @@ func ipsecMetricHandler(table string, model model.Model) {
 	}
 }
 
+// MontiorIPsecTunnelsState will register a labeled metric for the ipsec tunnel status. It will keep
+// checking ipsec established status for each Geneve tunnel towards other remote node for every 5s
+// and update the metric with tunnel up or down status.
+func MontiorIPsecTunnelsState(stopChan <-chan struct{}, wg *sync.WaitGroup, nbClient libovsdbclient.Client,
```

Comment:
did you look into adding event handles for this instead of polling? just curious.

Reply:
yes @martinkennelly , seems like `netlink.SubscribeAt(nl.NetnsNone(), nl.NetnsNone(), unix.NETLINK_XFRM, 0xffffffff)` can be used here, will try this out.